File: csharp/CleanTests.Tests/AntiPatterns/PriceEngineTests.cs
using System.Collections.Generic;
using CleanTests.Customer;
using FluentAssertions;
using LanguageExt;
using Xunit;

namespace CleanTests.Tests.AntiPatterns;

public class PriceEngineTests
{
    [Fact]
    public void Discount_Of_3_Products_Should_Be_3_Percent()
    {
        var products = Seq.create(new Product("P1"), new Product("P2"), new Product("P3"));
        var discount = PriceEngine.CalculateDiscount(products.ToArray());

        discount.Should().Be(products.Length * 0.01);
    }
}

File: php/tests/AntiPatterns/PriceEngineTests.php
<?php

namespace AntiPatterns;

use Com\CleanTests\Customer\PriceEngine;
use Com\CleanTests\Customer\Product;
use PHPUnit\Framework\TestCase;
use function PHPUnit\Framework\assertEquals;

class PriceEngineTests extends TestCase
{
    public function testDiscountOf3ProductsShouldBe3Percent(): void
    {
        $products = [new Product("P1"), new Product("P2"), new Product("P3")];
        $priceEngine = new PriceEngine();
        $discount = $priceEngine->calculateDiscount($products);

        assertEquals(count($products) * 0.01, $discount);
    }
}